File: code_blocks/integrations/third-party-integrations/firebase-integration_2.swift
import SwiftUI
import FirebaseAnalytics
import RevenueCat

@main
struct MyApp: App {
    init() {
        Purchases.configure(withAPIKey: "public_sdk_key")
        if let instanceID = Analytics.appInstanceID() {
            Purchases.shared.attribution.setFirebaseAppInstanceID(instanceID)
        }
    }

    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

File: code_blocks/integrations/third-party-integrations/firebase-integration_2.ts
import analytics from '@react-native-firebase/analytics';
import Purchases from 'react-native-purchases';

async function init() {
  Purchases.configure({ apiKey: 'public_sdk_key' });
  const instanceId = await analytics().getAppInstanceId();
  await Purchases.setFirebaseAppInstanceId(instanceId);
}
